src/main/java/tds/packager/mapper/ContentSpecFormatConverter.java

```
@@ -0,0 +1,116 @@
+            return Optional.empty();
+        }
+
+        if (isEnhanced(idString)) {
+            return Optional.of(idString);
+        }
+
+        try {
+            String fullContentId = idString.startsWith(prefix) ? idString : prefix + idString;
+            ContentSpecId id = converter.parse(fullContentId, grade);
+            return Optional.of(converter.format(id, ContentSpecFormat.ENHANCED));
+        } catch (ValidationException e) {
+            log.warn("Enhanced ID Issue: Cannot convert legacy ID to enhanced format '" + idString + "': "
+                    + e.getErrorDescription());
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Converts enhanced IDs to legacy format. If the passed in parameter is already in legacy format, or is a
+     * not a valid content spec ID at all, it will be returned without conversion.
+     *
+     * @param idString the ID to convert
+     * @return optional of the ID converted into legacy format, or left in legacy format if it was input that way.
+     * Returns Optional.empty() if the test's subject unspecified or unknown, or if there is a conversion error.
+     */
+    public static Optional<String> convertEnhancedToLegacy(String idString) {
+        if (idString == null) {
+            return Optional.empty();
+        }
+
+        if (!isEnhanced(idString)) {
+            return Optional.of(idString);
+        }
+
+        try {
+            if (converter.getFormatType(idString).isLegacy()) {
+                return Optional.of(idString);
+            }
+
+            ContentSpecId id = converter.parse(idString);
+            ContentSpecFormat targetFormat = id.isMath() ?  ContentSpecFormat.MATH_V6 : ContentSpecFormat.ELA_V1;
+            return Optional.of(converter.format(id, targetFormat));
+        } catch (ValidationException e) {
+            log.warn("Enhanced ID Issue: Cannot convert enhanced ID to legacy format '" + idString + "': "
```

If you move this to the library I feel that we should not catch any of these Exceptions and instead allow the caller to catch it if they simply want to warn.  For example, the converter.

Regardless, I think this should throw an Exception because the only way this will fail in this project is if the items have invalid legacy or enhanced ids.  I feel that is Exceptional and should stop the building of a test package.